catch/unit/texture/hipTexObjectTests.cc

```diff
@@ -0,0 +1,214 @@
+    memset(&array_desc, 0, sizeof(array_desc));
+    array_desc.Format = HIP_AD_FORMAT_FLOAT;
+    array_desc.NumChannels = 1;
+    array_desc.Width = width;
+    array_desc.Height = 0;
+
+    HIP_CHECK(hipArrayCreate(&array_member, &array_desc));
+    HIP_CHECK(hipMemcpyHtoA(reinterpret_cast<hipArray*>(array_member), 0, host_data_, size));
+
+    memset(&res_desc, 0, sizeof(res_desc));
+    res_desc.resType = HIP_RESOURCE_TYPE_ARRAY;
+    res_desc.res.array.hArray = array_member;
+    res_desc.flags = 0;
+
+    memset(&tex_desc, 0, sizeof(tex_desc));
+    tex_desc.filterMode = HIP_TR_FILTER_MODE_POINT;
+    tex_desc.flags = 0;
+
+    memset(&res_view_desc, 0, sizeof(res_view_desc));
+
+
+    if (useResourceViewDescriptor) {
+#if HT_AMD
+      res_view_desc.format = HIP_RES_VIEW_FORMAT_FLOAT_1X32;
+      res_view_desc.width = size;
+#else
+      /* Resource View Descriptors are not supported on NVIDIA currently */
+      useResourceViewDescriptor = false;
+#endif
+    }
+
+
+    HIP_CHECK(hipTexObjectCreate(&texture_object, &res_desc, &tex_desc,
+                                 useResourceViewDescriptor ? &res_view_desc : nullptr));
+  }
+
+  ~TexObjectTestWrapper() {
+    if (!ommit_destroy_) {
+      HIP_CHECK(hipTexObjectDestroy(texture_object));
+    }
+    HIP_CHECK(hipArrayDestroy(array_member));
+    free(host_data_);
+  }
+};
+
+/* hipTexObjectGetResourceDesc tests */
+
+TEST_CASE("Unit_hipGetTexObjectResourceDesc_positive") {
+  CHECK_IMAGE_SUPPORT;
+
+  TexObjectTestWrapper tex_obj_wrapper(false);
+
+  HIP_RESOURCE_DESC check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipTexObjectGetResourceDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.resType == tex_obj_wrapper.res_desc.resType);
+  REQUIRE(check_desc.res.array.hArray == tex_obj_wrapper.res_desc.res.array.hArray);
+}
+
+
+TEST_CASE("Unit_hipGetTexObjectResourceDesc_Negative_Parameters") {
+  CHECK_IMAGE_SUPPORT;
+
+  TexObjectTestWrapper tex_obj_wrapper(false);
+
+  HIP_RESOURCE_DESC check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  SECTION("desc is nullptr") {
+    HIP_CHECK_ERROR(hipTexObjectGetResourceDesc(nullptr, tex_obj_wrapper.texture_object),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(hipTexObjectGetResourceDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+                    hipErrorInvalidValue);
+  }
+}
+
+
+/* hipTexObjectGetResourceViewDesc tests */
+#if HT_AMD
+TEST_CASE("Unit_hipGetTexObjectResourceViewDesc_positive") {
+  CHECK_IMAGE_SUPPORT;
+
+  TexObjectTestWrapper tex_obj_wrapper(true);
+
+  HIP_RESOURCE_VIEW_DESC check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipTexObjectGetResourceViewDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.format == tex_obj_wrapper.res_view_desc.format);
+  REQUIRE(check_desc.width == tex_obj_wrapper.res_view_desc.width);
+}
+#endif
+
+#if HT_AMD
```

Comment:
can you just remove the test from json for nv?

Reply:
These tests have been modified to work on Nvidia and are enabled now.